qt_ui/windows/mission/flight/waypoints/QFlightWaypointList.py

```diff
@@ -27,17 +32,21 @@ def __init__(self, package: Package, flight: Flight):
             self.indexAt(QPoint(1, 1)), QItemSelectionModel.Select
         )
 
-    def update_list(self):
+    def update_list(self) -> None:
+        # disconnect itemChanged signal from handler slot for making altitude changes
+        # in advance updating waypoint list
+        self.model.itemChanged.disconnect(self.on_changed)
```

Comment:
You want something like this: https://github.com/dcs-liberation/dcs_liberation/blob/c45ac50370459a5d8109abac2e269bbbab3f9951/qt_ui/windows/mission/flight/SquadronSelector.py#L38-L42

Reply:
Cheers, changed to using blockSignals()